core/trino-spi/src/main/java/io/trino/spi/expression/Call.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.expression;
+
+import io.trino.spi.type.Type;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class Call
+        extends ConnectorExpression
+{
+    private final String name;
```

Comment:
This should be some form of qualified name. Expressions at the ConnectorExpression IR level should always operate on fully qualified names.

Since built-in functions don't (yet) have a catalog and schema qualifier (until https://github.com/trinodb/trino/issues/8 is completed), we should simply have an `Optional` catalog and schema for now. When they are missing, the function is interpreted as a built in function.

Reply:
I can't use the actual `QualifiedName` object here without adding a dependency to trino-parser. How can I create such a ‘form of’ qualified name? Can you please provide an example? 
I can't replace `String name` with catalog and schema `Optional`s because we need to infer which function `Call` represents. In addition, `Call` is created before the pushdown, so we still don't have the catalog. When should I populate the catalog and schema `Optional`s?

Reply:
Done